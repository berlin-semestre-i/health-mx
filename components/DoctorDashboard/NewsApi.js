import NewsAPI from 'newsapi'

const newsapi = new NewsAPI('350348a77322435f88f4310862eec6bd')

export default {
  getData() {
    return newsapi.v2.everything({
      q: 'salud',
      language: 'es',
      pageSize: 4
    })
  }
}