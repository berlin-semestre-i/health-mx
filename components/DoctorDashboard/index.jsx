import React from 'react'
import Layout from '../Layout'
import Header from '../shared/Header'
import { Grid, Card, Image, Button } from 'semantic-ui-react'
import styled from 'styled-components'
import Link from 'next/link'
import NewsApi from './NewsApi'

class DoctorDashboard extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      articles: []
    }
  }

  componentDidMount() {
    NewsApi.getData().then( response => {
      this.setState({
        articles: response.articles
      })
    })
  }

  render() {
    const { articles } = this.state
    return (
      <Layout userRole="medic">
        <Grid>
          <Header 
            title="Inicio" 
            subtitle="Buenos días, Dr. McDreamy"
          />
          <Grid.Row>
            <Grid.Column mobile={16} computer={8}>
              <CustomCard>
                <Card.Content>
                  <Card.Header>
                    Próximas Consultas
                  </Card.Header>
                    <Consultations>
                      <ConsultationRow>
                        <Grid.Column width={2}>
                          <Avatar src="../../static/images/avatar.png"/>
                        </Grid.Column>
                        <Grid.Column width={14}>
                          <Link as="" href=""><a>María del Carmen Hinojosa Ramírez</a></Link>
                          <ConsultationInfo>
                            <span>
                              <Bold>Fecha: </Bold> 10 Oct, 2048
                            </span>
                            <span>
                              <Bold>Hora: </Bold> 08:30 am
                            </span>
                            <span>
                              <Bold>Última cita: </Bold> 12 Jul, 2048
                            </span>
                          </ConsultationInfo>
                        </Grid.Column>
                      </ConsultationRow>
                  </Consultations>
                  <CenteredButton>
                    <MoreButton secondary className="small">Ver más</MoreButton>
                  </CenteredButton> 
                </Card.Content>
              </CustomCard>
              <CustomCard>
                <Card.Content>
                  <Card.Header>
                    Información Importante
                  </Card.Header>
                  <p>
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vel sollicitudin est, at varius nunc. 
                    Morbi ac quam at est viverra posuere. In porttitor tortor vulputate dolor semper egestas. Curabitur sit 
                    amet ante sed erat tempor pellentesque. Pellentesque ipsum lectus, condimentum a volutpat id, tincidunt 
                    vitae nunc. Ut sed dapibus mi, eu finibus arcu. Nulla maximus posuere nunc id accumsan. Pellentesque tempor 
                    aliquam lectus, quis efficitur nisl suscipit vel. Pellentesque vitae dolor eros. 
                  </p>
                </Card.Content>
              </CustomCard>
            </Grid.Column>
            <Grid.Column mobile={16} computer={8}>
              <CustomCard>
                <Card.Content>
                  <Card.Header>
                    Últimas Noticias
                  </Card.Header>
                  {articles.map((article, index) => (
                    <ArticleContainer key={index}>
                      <p>{article.title}</p>
                      <a href={article.url} target="_blank">
                        <MoreButton secondary className="small">Ver más</MoreButton>
                      </a>
                    </ArticleContainer>
                  ))}
                </Card.Content>
              </CustomCard>
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </Layout>
    )
  }
}

export default DoctorDashboard

const MoreButton = styled(Button)`
  &&& {
    padding: .5em 1em;
  }
`

const ArticleContainer = styled.div`
  padding: 1.5rem 0;

  :not(:last-child) {
    border-bottom: 1px solid rgba(0,0,0,0.1);
  }
`

const CenteredButton = styled.div`
  text-align: center;
`

const Consultations = styled.div`
  margin-bottom: 1rem;
`

const CustomCard = styled(Card)`
  && {
    width: 100%;
  }
`
const ConsultationRow = styled(Grid)`
  >.column:not(.row):last-child {
    padding-top: 1.5rem;
  }
`

const ConsultationInfo = styled.div`
  width: 85%;
  display: flex;
  justify-content: space-between;
  font-size: 12px;
`
const Avatar = styled(Image)`
  && {
    border-radius: 30px;
    max-width: 60px;
    max-height: 60px;
    box-shadow: 0 1px 4px rgba(0, 0, 0, 0.3);
    -webkit-box-shadow: 0 1px 4px rgba(0, 0, 0, 0.3);
    border: 3px solid #FFF;
  }
`
const Bold = styled.b`
  font-weight: 500;
`