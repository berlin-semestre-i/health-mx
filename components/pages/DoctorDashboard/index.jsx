import React, { PureComponent } from 'react'
import Layout from '../../Layout'
import Header from '../../shared/Header'
import AppointmentListItem from '../../shared/AppointmentListItem'
import Card from '../../shared/Card'
import { Grid, Button, Loader, Dimmer } from 'semantic-ui-react'
import styled from 'styled-components'
import NewsApi from './NewsApi'
import Link from 'next/link'

class DoctorDashboard extends PureComponent {

  state = {
    articles: [],
    appointment: {
      name: "María del Carmen Hinojosa Ramírez",
      date: "10 Oct, 2018",
      time: "08:30 am",
      previous: "12 Jul, 2048",
      profile: "../../../static/images/avatar.png"
    }
  }

  componentDidMount() {
    NewsApi.getData().then( response => {
      this.setState({
        articles: response.articles
      })
    })
  }

  render() {
    const { articles, appointment } = this.state
    
    return (
      <React.Fragment>
        <Layout userRole="medic">
          <Grid>
            <Header 
              title="Inicio" 
              subtitle="Buenos días, Dr. McDreamy"
            />
            <Grid.Row>
              <Grid.Column mobile={16} tablet={16} computer={8} className="dashboard-col">
                <Card header="Próximas Consultas">
                  <Consultations>
                      <AppointmentListItem appointment={appointment}/>
                  </Consultations>
                  <CenteredButton>
                    <MoreButton secondary className="small">Ver más</MoreButton>
                  </CenteredButton> 
                </Card>
                <Card header="Información Importante">
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vel sollicitudin est, at varius nunc. 
                  Morbi ac quam at est viverra posuere. In porttitor tortor vulputate dolor semper egestas. Curabitur sit 
                  amet ante sed erat tempor pellentesque. Pellentesque ipsum lectus, condimentum a volutpat id, tincidunt 
                  vitae nunc. Ut sed dapibus mi, eu finibus arcu. Nulla maximus posuere nunc id accumsan. Pellentesque tempor 
                  aliquam lectus, quis efficitur nisl suscipit vel. Pellentesque vitae dolor eros. 
                </Card>
              </Grid.Column>
              <Grid.Column mobile={16} tablet={16} computer={8}>
                <Card header="Últimas Noticias" className="loaderCard nth-col-first">
                  <NewsDimmer active={articles.length == 0}>
                    <Loader indeterminate />
                  </NewsDimmer>
                  {articles.map((article, index) => (
                    <ArticleContainer name="article" key={index}>
                      <p>{article.title}</p>
                      <Link href={article.url} >
                        <a target="_blank">
                          <MoreButton secondary className="small">Ver más</MoreButton>
                        </a>
                      </Link>
                    </ArticleContainer>
                  ))}
                </Card>
              </Grid.Column>
            </Grid.Row>
          </Grid>
        </Layout>
      </React.Fragment>
    )
  }
}

export default DoctorDashboard

const NewsDimmer = styled(Dimmer)`
  && {
    background-color: rgba(0,0,0,0);
  }
`
const MoreButton = styled(Button)`
  &&& {
    padding: .5em 1em;
  }
`
const ArticleContainer = styled.div`
  padding: 1.5rem 0;

  :not(:last-child) {
    border-bottom: 1px solid rgba(0,0,0,0.1);
  }
`
const CenteredButton = styled.div`
  text-align: center;
`
const Consultations = styled.div`
  margin-bottom: 1rem;
`