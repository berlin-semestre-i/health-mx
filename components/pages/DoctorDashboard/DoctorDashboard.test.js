import React from 'react'
import { shallow, mount } from 'enzyme'
import DoctorDashboard from './index'

describe('DoctorDashboard', () => {
  it('renders', () => {
    const component = shallow(<DoctorDashboard />)
    expect(component).toMatchSnapshot()
  })

  it('has three cards', () => {
    const wrapper = mount(<DoctorDashboard />)
    expect(wrapper.find('div.card').length).toBe(3)
    wrapper.unmount()
  })

  // it('has four articles', async () => {
  //  const wrapper = mount(<DoctorDashboard />)
  //  wait(5000)
  //  console.log(wrapper.find('[name="article"]').debug())
  //  expect(wrapper.find('div[name="article"]').length).toBe(4);
  //  wrapper.unmount();
  // })
})