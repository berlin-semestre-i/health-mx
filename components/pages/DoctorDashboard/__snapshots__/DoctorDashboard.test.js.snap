// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoctorDashboard renders 1`] = `
<Fragment>
  <Layout
    userRole="medic"
  >
    <Grid>
      <Header
        subtitle="Buenos días, Dr. McDreamy"
        title="Inicio"
      />
      <GridRow>
        <GridColumn
          className="dashboard-col"
          computer={8}
          mobile={16}
          tablet={16}
        >
          <CustomCard
            header="Próximas Consultas"
          >
            <DoctorDashboard__Consultations>
              <AppointmentListItem
                appointment={
                  Object {
                    "date": "10 Oct, 2018",
                    "name": "María del Carmen Hinojosa Ramírez",
                    "previous": "12 Jul, 2048",
                    "profile": "../../../static/images/avatar.png",
                    "time": "08:30 am",
                  }
                }
              />
            </DoctorDashboard__Consultations>
            <DoctorDashboard__CenteredButton>
              <DoctorDashboard__MoreButton
                className="small"
                secondary={true}
              >
                Ver más
              </DoctorDashboard__MoreButton>
            </DoctorDashboard__CenteredButton>
          </CustomCard>
          <CustomCard
            header="Información Importante"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vel sollicitudin est, at varius nunc. Morbi ac quam at est viverra posuere. In porttitor tortor vulputate dolor semper egestas. Curabitur sit amet ante sed erat tempor pellentesque. Pellentesque ipsum lectus, condimentum a volutpat id, tincidunt vitae nunc. Ut sed dapibus mi, eu finibus arcu. Nulla maximus posuere nunc id accumsan. Pellentesque tempor aliquam lectus, quis efficitur nisl suscipit vel. Pellentesque vitae dolor eros.
          </CustomCard>
        </GridColumn>
        <GridColumn
          computer={8}
          mobile={16}
          tablet={16}
        >
          <CustomCard
            className="loaderCard nth-col-first"
            header="Últimas Noticias"
          >
            <DoctorDashboard__NewsDimmer
              active={true}
            >
              <Loader
                indeterminate={true}
              />
            </DoctorDashboard__NewsDimmer>
          </CustomCard>
        </GridColumn>
      </GridRow>
    </Grid>
  </Layout>
</Fragment>
`;
